process/smartContract/scQueryService.go

```diff
@@ -88,6 +95,19 @@ func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice ui
 	log.Trace("executeScCall", "function", query.FuncName, "numQueries", service.numQueries)
 	service.numQueries++
 
+	shouldEarlyExitBecauseOfSyncState := query.ShouldBeSynced && service.bootstrapper.GetNodeState() == common.NsNotSynchronized
+	if shouldEarlyExitBecauseOfSyncState {
```

Comment:
```suggestion
	nodeNotSynced := service.bootstrapper.GetNodeState() == common.NsNotSynchronized
	if query.ShouldBeSynced && nodeNotSynced {
```

Reply:
I don't think it's best because if the flag isn't set, we don't care about the sync state. I would leave it as it is